File: behavioral_patterns/template_method/typescript/src/Tea.ts
import {BeverageProductionTemplate} from "./BeverageProductionTemplate";

export class Tea extends BeverageProductionTemplate {
  protected mixIngredients(): void {
      console.log("Steeping the tea");
  }

  protected addCondiments(): void {
      console.log("Adding Lemon");
  }
}

File: behavioral_patterns/template_method/java/src/beverageProductionTemplate/Tea.java
package beverageProductionTemplate;

public class Tea extends BeverageProductionTemplate {

    @Override
    void mixIngredients() {
        System.out.println("Steeping the tea");
    }

    @Override
    void addCondiments() {
        System.out.println("Adding Lemon");
    }
}
